src/controllers/message.controller.ts

```diff
@@ -0,0 +1,21 @@
+import { RequestWithUser } from '@/interfaces';
+import { NextFunction, Response, Request } from 'express';
+import { MessageService } from '@/services';
+import { ObjectId } from 'mongoose';
+import { BaseMessageDto, CreateMessageDto } from '@/dtos';
+
+class MessageController {
+  private messageService = new MessageService();
+
+  public getChatroomMessages = async (req: Request, res: Response, next: NextFunction) => {
+    try {
+      const id = req.params.chat_id as unknown as ObjectId;
```

Comment:
what does this mean?

Reply:
casting string type chat_id to mongoose's built in ObjectId type, which is mongodb's "primary key".